src/Gui/Action.cpp

```diff
@@ -539,6 +539,34 @@ void WorkbenchGroup::addTo(QWidget *w)
         connect(_group, SIGNAL(triggered(QAction*)), box, SLOT(onActivated (QAction*)));
         bar->addWidget(box);
     }
+    else if (w->inherits("QMenuBar")) {
+        auto menuBar = qobject_cast<QMenuBar*>(w);
+        ParameterGrp::handle hGrp = WindowParameter::getDefaultParameter()->GetGroup("General");
+
+        if (hGrp->GetBool("WBSLeftCorner", true)) {
+            QComboBox* box = new WorkbenchComboBox(this, w);
+            box->setIconSize(QSize(16, 16));
+            box->setToolTip(_action->toolTip());
+            box->setStatusTip(_action->statusTip());
+            box->setWhatsThis(_action->whatsThis());
+            box->addActions(_group->actions());
+            connect(_group, SIGNAL(triggered(QAction*)), box, SLOT(onActivated(QAction*)));
+            menuBar->setCornerWidget(box, Qt::TopLeftCorner);
+            box->show();
+        }
+        if (hGrp->GetBool("WBSRightCorner", true)) {
+            QComboBox* box = new WorkbenchComboBox(this, w);
+            box->setIconSize(QSize(16, 16));
+            box->setToolTip(_action->toolTip());
+            box->setStatusTip(_action->statusTip());
+            box->setWhatsThis(_action->whatsThis());
+            box->addActions(_group->actions());
+            connect(_group, SIGNAL(triggered(QAction*)), box, SLOT(onActivated(QAction*)));
+            menuBar->setCornerWidget(box, Qt::TopRightCorner);
+            box->show();
+        }

```

Comment:
This is almost twice the same code, there is probably better to do.
Also, is that really an option that someone wants this on both corners ?

Reply:
Indeed. Besides it will change as we settled for a combobox offering either left or right corner.